src/nav_msg.c

```diff
@@ -228,15 +228,8 @@ s8 process_subframe(gnss_signal_t sid, nav_msg_t *n,
 {
   // Check parity and parse out the ephemeris from the most recently received subframe
 
-  if (!data) {
-    log_error("process_subframe: CALLED WITH e = NULL!");
-    n->subframe_start_index = 0;  // Mark the subframe as processed
-    n->next_subframe_id = 1;      // Make sure we start again next time
-    return -1;
-  }
-
-  data->ephemeris_upd_flag = false;
-  data->gps_l2c_sv_capability_upd_flag = false;
+  assert(data != NULL);
+  memset(data, 0, sizeof(gps_l1ca_decoded_data_t));
```

Comment:
I think only setting the flags to false was fine. Also, at present the sid field is initialized externally.